### PR TITLE
Replace ping with curl

### PR DESCRIPTION
**Pull request by AlexandreDey**
Some network have HTTP(S) and DNS connectivity with internet but not ICMP.
Using ping, the connectivity check returns False, even though it might be possible to connect.
This PR changes ping to curl, to check for HTTP connectivity with the servers instead of ICMP.

### CODE REVIEW

File: src/nvautoinstall/MainFunction.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def avbl(self):
         return "rpmfusion-nonfree-nvidia-driver" in output
 
     def conn(self):
-        retndata = subprocess.getstatusoutput("ping -c 3 -W 3 rpmfusion.org")[0]
```
Comment (gridhead) [inline src/nvautoinstall/MainFunction.py:115]: We'd definitely not want to depend on ICMP connectivity on this.

```
ping -c 3 -W 3 developer.download.nvidia.com
```
```
PING cs500.wpc.phicdn.net (152.199.39.144) 56(84) bytes of data.
64 bytes from 152.199.39.144 (152.199.39.144): icmp_seq=1 ttl=60 time=9.21 ms
64 bytes from 152.199.39.144 (152.199.39.144): icmp_seq=2 ttl=60 time=9.47 ms
64 bytes from 152.199.39.144 (152.199.39.144): icmp_seq=3 ttl=60 time=9.34 ms

--- cs500.wpc.phicdn.net ping statistics ---
3 packets transmitted, 3 received, 0% packet loss, time 2002ms
rtt min/avg/max/mdev = 9.209/9.341/9.472/0.107 ms
```

File: src/nvautoinstall/MainFunction.py [hunk 2]
```diff
@@ -112,7 +112,7 @@ def avbl(self):
...
+        retndata = subprocess.getstatusoutput("curl rpmfusion.org")[0]
```
Comment (gridhead) [inline src/nvautoinstall/MainFunction.py]: The usage with CURL returns the following output.
```
curl rpmfusion.org
```
```
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>302 Found</title>
</head><body>
<h1>Found</h1>
<p>The document has moved <a href="https://rpmfusion.org/">here</a>.</p>
<hr>
<address>Apache Server at rpmfusion.org Port 80</address>
</body></html>
```

I suggest it be changed to the following.
```suggestion
        retndata = subprocess.getstatusoutput("curl https://rpmfusion.org/")[0]
```

File: src/nvautoinstall/MainFunction.py [hunk 3]
```diff
@@ -156,7 +156,7 @@ def rpin(self):
         return retndata == 0
 
     def conn(self):
-        retndata = subprocess.getstatusoutput("ping -c 3 -W 3 developer.download.nvidia.com")[0]
```
Comment (gridhead) [inline src/nvautoinstall/MainFunction.py:159]: We'd definitely not want to depend on ICMP connectivity on this.
```
ping -c 3 -W 3 developer.download.nvidia.com
```
```
PING cs500.wpc.phicdn.net (152.199.39.144) 56(84) bytes of data.
64 bytes from 152.199.39.144 (152.199.39.144): icmp_seq=1 ttl=60 time=12.0 ms
64 bytes from 152.199.39.144 (152.199.39.144): icmp_seq=2 ttl=60 time=9.35 ms
64 bytes from 152.199.39.144 (152.199.39.144): icmp_seq=3 ttl=60 time=10.8 ms

--- cs500.wpc.phicdn.net ping statistics ---
3 packets transmitted, 3 received, 0% packet loss, time 2002ms
rtt min/avg/max/mdev = 9.345/10.698/11.992/1.081 ms
```

File: src/nvautoinstall/MainFunction.py [hunk 4]
```diff
@@ -156,7 +156,7 @@ def rpin(self):
...
+        retndata = subprocess.getstatusoutput("curl developer.download.nvidia.com")[0]
```
Comment (gridhead) [inline src/nvautoinstall/MainFunction.py]: The usage with CURL returns the following output.
```
curl developer.download.nvidia.com
```
```
<?xml version="1.0" encoding="iso-8859-1"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
         "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
	<head>
		<title>404 - Not Found</title>
	</head>
	<body>
		<h1>404 - Not Found</h1>
		<script type="text/javascript" src="//wpc.75674.betacdn.net/0075674/www/ec_tpm_bcon.js"></script>
	</body>
</html>
```

I suggest it be changed to the following.
```suggestion
        retndata = subprocess.getstatusoutput("curl https://developer.download.nvidia.com/compute/cuda/repos/")[0]
```

**Review by gridhead — CHANGES_REQUESTED** [review]
Thank you for your pull request.

Please make the suggested changes and we can go ahead with a merge.

File: src/nvautoinstall/MainFunction.py (hunk 4 above)
Comment (AlexandreDey) [reply src/nvautoinstall/MainFunction.py]: Thanks for the suggestion ! 
Switched to https for rpmfusion.org and the more specific url for nvidia's repo in the latest [commit](https://github.com/t0xic0der/nvidia-auto-installer-for-fedora/pull/66/commits/aa8295e6468fce7fa91b4a6a5278ed0f852cfb9a).

**Review by gridhead — APPROVED** [review]
Looks good to me.

**Pull request merged**